services/src/main/java/org/fao/geonet/api/records/backup/DownloadBackup.java

```diff
@@ -0,0 +1,124 @@
+
+@RequestMapping(value = {
+        "/api/records",
+        "/api/" + API.VERSION_0_1 +
+                "/records"
+})
+@Api(value = API_CLASS_RECORD_TAG,
+        tags = API_CLASS_RECORD_TAG,
+        description = API_CLASS_RECORD_OPS)
+
+@Controller
+public class DownloadBackup {
+
+    @Autowired
+    ServiceManager serviceManager;
+
+    @Autowired
+    ArchiveAllMetadataJob archiveAllMetadataJob;
+
+    @ApiOperation(value = "Download MEF backup archive",
+            notes = ".",
+            nickname = "downloadBackup")
+    @RequestMapping(value="/download.backup",produces = "application/zip")
+    @ResponseBody
+    public ResponseEntity<FileSystemResource> exec(HttpServletRequest request) throws Exception {
+
+        ServiceContext context = ApiUtils.createServiceContext(request);
+        ApplicationContext appContext = ApplicationContextHolder.get();
+        GeonetworkDataDirectory dataDirectory = appContext.getBean(GeonetworkDataDirectory.class);
+
+        Log.info(ArchiveAllMetadataJob.BACKUP_LOG, "User " + context.getUserSession().getUsername() + " from IP: " + context
+                .getIpAddress() + " has started to download backup archive");
+
+        File backupDir = dataDirectory.getBackupDir().resolve(ArchiveAllMetadataJob.BACKUP_DIR).toFile();
+        if (!backupDir.exists()) {
+            throw404();
+        }
+        File[] files = backupDir.listFiles(new FileFilter() {
+            @Override
+            public boolean accept(File pathname) {
+                return !pathname.isDirectory();
+            }
+        });
+        if (files == null || files.length == 0) {
+            throw404();
+        }
+        MultiValueMap<String, String> headers = new HttpHeaders();
+        headers.add("content-disposition", "attachment; filename=" + files[0].getName());
+
+        final ResponseEntity<FileSystemResource> response = new ResponseEntity<>(new FileSystemResource(files[0]), headers, HttpStatus.OK);
+
+        return response;
+    }
+
+    @ApiOperation(value = "Trigger MEF backup archive",
+            notes = ".",
+            nickname = "triggerBackup")
+    @RequestMapping(
+            value="/trigger.backup",
+            produces = MediaType.APPLICATION_JSON_VALUE
+    )
+    @PreAuthorize("hasRole('Administrator')")
+    @ResponseStatus(HttpStatus.OK)
+    @ApiResponses(value = {
+            @ApiResponse(code = 200, message = "Return succeed message.")
+    })
+    @ResponseBody
+    public String trigger(HttpServletRequest request) throws Exception {
+        ServiceContext context = ApiUtils.createServiceContext(request);
+        final Trigger trigger = newTrigger().forJob("archiveAllMetadata", "gnBackgroundTasks").startNow().build();
+        context.getApplicationContext().getBean("gnBackgroundJobScheduler", Scheduler.class).scheduleJob(trigger);
+
+        return "{\"success\":true}";
+    }
+
+    private void throw404() throws JeevesException {
```

Comment:
I think you could use `ResourceNotFoundException` instead, see in https://github.com/geonetwork/core-geonetwork/blob/3.4.x/services/src/main/java/org/fao/geonet/api/csw/VirtualCswApi.java#L87-L106